### PR TITLE
chore(auth): cleanup and convert to kotlin device srp helper

### PR DESCRIPTION
**Pull request by div5yesh**
- [ ] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [ ] Added Unit Tests
- [ ] Added Integration Tests

*Documentation update required?*
- [ ] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/helpers/CognitoDeviceHelperTests.kt [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
 
 class CognitoDeviceHelperTests {
-
-    @Test
-    fun verifyHexN() {
-        val expectedHex = "FFFFFFFFFFFFFFFFC90FDAA22168C234C4C6628B80DC1CD129024E088A67CC74020BBEA63B139B22514A08798" +
-            "E3404DDEF9519B3CD3A431B302B0A6DF25F14374FE1356D6D51C245E485B576625E7EC6F44C42E9A637ED6B0BFF5CB6F40" +
-            "6B7EDEE386BFB5A899FA5AE9F24117C4B1FE649286651ECE45B3DC2007CB8A163BF0598DA48361C55D39A69163FA8FD24C" +
-            "F5F83655D23DCA3AD961C62F356208552BB9ED529077096966D670C354E4ABC9804F1746C08CA18217C32905E462E36CE" +
-            "3BE39E772C180E86039B2783A2EC07A28FB5C55DF06F4C52C9DE2BCBF6955817183995497CEA956AE515D2261898FA05" +
```
Comment (gpanshu) [inline aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/helpers/CognitoDeviceHelperTests.kt:14]: Why are we removing this test?

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/events/SignInEvent.kt [hunk 2]
```diff
@@ -49,6 +49,7 @@ class SignInEvent(val eventType: EventType, override val time: Date? = null) : S
         data class InitiateHostedUISignIn(val hostedUISignInData: SignInData.HostedUISignInData) : EventType()
         data class SignedIn(val id: String = "") : EventType()
         data class ConfirmDevice(
+            val username: String,
```
Comment (gpanshu) [inline aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/events/SignInEvent.kt:52]: Username is part of signed in data. Why do we need to add it here?

**Review by gpanshu — CHANGES_REQUESTED** [review]
Please review my comments before merging

File: aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/helpers/CognitoDeviceHelperTests.kt (hunk 1 above)
Comment (tylerjroach) [reply aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/helpers/CognitoDeviceHelperTests.kt:14]: Looks like the method was removed in favor of using private string of same value. I'm ok with this change.

**Review by tylerjroach — APPROVED** [review]
See @gpanshu's comment on username, but otherwise approved.

**Review by gpanshu — APPROVED** [review]

**Pull request merged**